### PR TITLE
Embarking penalty logic fix

### PR DESCRIPTION
**Pull request by itanasi**

### CODE REVIEW

File: core/src/com/unciv/logic/battle/BattleDamage.kt [hunk 1]
```diff
@@ -136,11 +136,11 @@ object BattleDamage {
                 modifiers["Landing"] = -50
 
             // Land Melee Unit attacking to Water
-            if (!attacker.unit.isEmbarked() && attacker.isMelee() && defender.getTile().isWater
+            if (attacker.unit.type.isLandUnit() && !attacker.unit.isEmbarked() && attacker.isMelee() && defender.getTile().isWater
                     && !attacker.unit.hasUnique(UniqueType.AttackAcrossCoast))
                 modifiers["Boarding"] = -50
             // Naval Unit Melee attacking to Land (not City) unit
-            if (attacker.unit.type.isWaterUnit() && attacker.isMelee() && !defender.getTile().isWater
+            if (attacker.unit.type.isWaterUnit() && attacker.isMelee() && !defender.getTile().isLand
```
Comment (xlenstra) [inline core/src/com/unciv/logic/battle/BattleDamage.kt:143]: I'm reasonably sure this was already the right way around

Comment (itanasi) [reply core/src/com/unciv/logic/battle/BattleDamage.kt:143]: You're right. Need to get some time to actually test

**Comment by yairm210** [comment]
Looks like it should be fine, is there a reason this is still a draft?

**Review by xlenstra — APPROVED** [review]

**Review by yairm210 — APPROVED** [review]

**Comment by itanasi** [comment]
Just haven't had a chance to robustly check it

**Comment by itanasi** [comment]
OK looks good now

**Pull request merged**